src/app

```diff
@@ -0,0 +1,131 @@
+#!/bin/sh
+
+set -e
+
+temp_dir=$(mktemp -d "app.tmp.XXXXXXXX")
+
+cleanup() {
+    set +e
+    test "$temp_dir" != "" && test "$temp_dir" != "/" && rm -Rf "${temp_dir}"
+    test -f "$persistent_store"/.rw_test && rm -f "$persistent_store"/.rw_test
+}
+
+trap cleanup 1 2 3 6 15
+
+config_file="/etc/mender/mender-app.conf"
+version=""
+orchestrator=""
+platform=""
+manifests_dir=""
+app_module_dir="/usr/share/mender/app-modules/v1"
+app_sub_module=""
+artifact_name=""
+persistent_store="/data/mender-app"
+application_name=""
+
+if test -f "$config_file"; then
+    . "$config_file"
+fi
+
+if test -d "$persistent_store"; then
+    if touch "$persistent_store"/.rw_test; then
+        rm -f "$persistent_store"/.rw_test
+    else
+        echo "ERROR: cant write to persistent_store in $persistent_store"
+        exit 1
+    fi
+fi
+
+delta_cmd="xdelta3 -d -s"
+tar_decompress_cmd="tar -xzvf"
+
+parse_metadata() {
+    application_name=$(jq -r .application_name < "$1")
+    platform=$(jq -r .platform < "$1")
+    orchestrator=$(jq -r .orchestrator < "$1")
+    version=$(jq -r .version < "$1")
+    app_sub_module="${app_module_dir}/${orchestrator}"
+    if test ! -f "${app_sub_module}"; then
+        echo "ERROR: ${app_sub_module} not found. exiting."
+        return 1
+    fi
+    artifact_name=$(jq -r .artifact_provides.artifact_name < "$2")
+}
+
+decompress_artifact() {
+    local image_dir
+    local image
+    local url_new
+    local url_current
+    local sha_new
+    local sha_current
+    local current_image
+    local new_image
```

Comment:
I moved to `image_new` and `image_current`. perhaps it will be ok, since it is only the output path where we save the binary data of the application components. I wanted to move from image, but it is not a bad name.

Reply:
It is not a bad name, it is only incomplete in my opinion. Because the other variables (url, sha) also refer to images (image url, image sha).

Is okey, though. Not a big deal.